### PR TITLE
add standard text id management

### PR DESCRIPTION
**Pull request by adamkorynta**
- [x] DAO class
- [x] Controller class
- [x] object serialization
- [x] Controller integration tests

### CODE REVIEW

**Comment by adamkorynta** [comment]
Based on: https://github.com/USACE/cwms-data-api/pull/470

**Comment by rma-rripken** [comment]
Looks good to me.  I'm not going to approve it yet b/c I think #470 has to go first...

File: cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java]: should still be a constant

File: cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java]: should still be a constant.

Comment (adamkorynta) [reply cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java]: added

File: cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java
Comment (adamkorynta) [reply cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java]: added

**Comment by adamkorynta** [comment]
Un-drafted, but still blocked by #470

**Review by rma-rripken — DISMISSED** [review]

File: cwms-data-api/src/main/java/cwms/cda/ApiServlet.java
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/ApiServlet.java]: This might read better with string formatting so the pull "path" is at least inline.

File: cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java:190]: Are we planing to implement this? Seems usual for things like: I made a typo.

File: cwms-data-api/src/test/java/cwms/cda/api/DataApiTestIT.java
Comment (MikeNeilson) [inline cwms-data-api/src/test/java/cwms/cda/api/DataApiTestIT.java]: you copy pasted my typo. `find for` -> `fine for`

File: cwms-data-api/src/test/java/cwms/cda/data/dao/TimeSeriesTextDaoTestIT.java
Comment (MikeNeilson) [inline cwms-data-api/src/test/java/cwms/cda/data/dao/TimeSeriesTextDaoTestIT.java]: I feel like this class name should indicate that it's the base of other test. But saying that it's true of DataApiTestIT itself.

**Review by MikeNeilson — DISMISSED** [review]

File: cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java
Comment (adamkorynta) [reply cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java:190]: Would the update revise the id or the value (or both)? I'd have to also add a PL/SQL procedure for the update as there isn't one currently. 

For this PR I only implemented what the existing Java client code was using and given a new procedure would be needed I'd do that in a separate PR.

**Review by MikeNeilson — COMMENTED** [review]
This otherwise looks okay, but it's difficult to review given that it appears to include the feature/text-time-series branch.

In the future when doing that open a PR to the derived from branch to make review easier.

Will continue review once #470 is merged in.

**Comment by MikeNeilson** [comment]
Needs a rebase/conflict fix. but otherwise still looks good.

File: cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java
Comment (MikeNeilson) [reply cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java:190]: Fair for "update" I would say just the value. a rename like that is enough of an operation to justify being it's own endpoint, I think.

Comment (MikeNeilson) [reply cwms-data-api/src/main/java/cwms/cda/api/StandardTextController.java:190]: But we can worry about that later.

**Review by MikeNeilson — DISMISSED** [review]

File: cwms-data-api/src/main/java/cwms/cda/ApiServlet.java
Comment (adamkorynta) [reply cwms-data-api/src/main/java/cwms/cda/ApiServlet.java]: updated

File: cwms-data-api/src/test/java/cwms/cda/api/DataApiTestIT.java
Comment (adamkorynta) [reply cwms-data-api/src/test/java/cwms/cda/api/DataApiTestIT.java]: I think this was Ryan's PR....

File: cwms-data-api/src/test/java/cwms/cda/data/dao/TimeSeriesTextDaoTestIT.java
Comment (adamkorynta) [reply cwms-data-api/src/test/java/cwms/cda/data/dao/TimeSeriesTextDaoTestIT.java]: I don't think this is the base for anything, but also wasn't added in this PR

**Review by MikeNeilson — APPROVED** [review]

File: cwms-data-api/src/test/java/cwms/cda/api/DataApiTestIT.java
Comment (MikeNeilson) [reply cwms-data-api/src/test/java/cwms/cda/api/DataApiTestIT.java]: Entirely Possible. It was just in this PR that I caught it.

File: cwms-data-api/src/test/java/cwms/cda/data/dao/TimeSeriesTextDaoTestIT.java
Comment (MikeNeilson) [reply cwms-data-api/src/test/java/cwms/cda/data/dao/TimeSeriesTextDaoTestIT.java]: Yep.

**Pull request merged**
